### DISCUSSION TITLE
tutorials/voyage-part1/07-how-the-router-resolves-data

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#180] [course feedback]
# tutorials/voyage-part1/07-how-the-router-resolves-data

Learn how the router creates query plans for resolving client operations across subgraphs

https://www.apollographql.com/tutorials/voyage-part1/07-how-the-router-resolves-data

**Comment by stephan-buckmaster** [comment]
Too much to ask? Can you link to the line in the source code of Apollo Router Core  where the router "uses this information to build a query plan," Some file/line under https://github.com/apollographql/router/tree/dev/apollo-router I am guessing

**Reply by lizhennessy** [reply]
Hi there @stephan-buckmaster  - the query planning part of the router is a pretty robust feature, so unfortunately I can't point to a single line where this occurs in the source code. However we have our [documentation section](https://www.apollographql.com/docs/graphos/reference/federation/query-plans) that goes into a lot more detail about how this process is completed!

**Reply by stephan-buckmaster** [reply]
Looks like I ventured into the 'wishful thinking' department with my question! 😄 Thanks for the guidance and pointing me to the documentation. I appreciate the help!
